### ISSUE TITLE
KeyError when loading a trained EncoderDecoder model

### ISSUE THREAD

**Issue by gustavscholin** [huggingface/transformers#4595]
# 🐛 Bug

## Information

Error when loading a trained EncoderDecoder model. When loading the config the in `configuration_auto.py` the `model_type` is expected on the form `encoder-decoder` but in   `configuration_encoder_decoder.py` `model_type` is on the form `encoder_decoder` which raises a KeyError.

The hyphen version seems to be convention in the other model configuration files.

I guess this is something for @patrickvonplaten

**Comment by patrickvonplaten** [comment]
Hi @gustavscholin, 

Thanks for you issue!
Could you please provide a code example so that I can reproduce the error?

**Comment by eunji6546** [comment]
Hi, @patrickvonplaten @gustavscholin 
For me, setting "base_model_prefix" in modeling_encoder_decoder.py fixed this problem, as finding params is based on self.base_model_prefix. 

Is it fundamental solution? or just short-sighted?

**Comment by gustavscholin** [comment]
@patrickvonplaten, here's a colab notebook to reproduce the error:

https://colab.research.google.com/drive/102U7pJJcyw__Yq0PERxAKvKPSx3bvNSi?usp=sharing

**Comment by patrickvonplaten** [comment]
> Hi, @patrickvonplaten @gustavscholin
> For me, setting "base_model_prefix" in modeling_encoder_decoder.py fixed this problem, as finding params is based on self.base_model_prefix.
> 
> Is it fundamental solution? or just short-sighted?

No that was the right solution :-) I did exactly the same in this fix:  #4680

**Comment by patrickvonplaten** [comment]
> @patrickvonplaten, here's a colab notebook to reproduce the error:
> 
> https://colab.research.google.com/drive/102U7pJJcyw__Yq0PERxAKvKPSx3bvNSi?usp=sharing

A saved encoder-decoder model will always only be saved in a single folder. A single folder can always be loaded with `.from_pretrained()`. So to make your notebook work, you simply have to replace this line:

```python 
saved_model = EncoderDecoderModel.from_encoder_decoder_pretrained('test_run', 'test_run')
```

by 

```python 
saved_model = EncoderDecoderModel.from_pretrained('test_run')
```
